nucleus/payara-modules/healthcheck-core/src/main/java/fish/payara/nucleus/healthcheck/preliminary/HoggingThreadsHealthCheck.java

```diff
@@ -91,65 +142,107 @@ public String getDescription() {
 
     @Override
     protected HealthCheckResult doCheckInternal() {
-        hoggingThreads.set(0);
         HealthCheckResult result = new HealthCheckResult();
-        ThreadMXBean threadBean = ManagementFactory.getThreadMXBean();
-
-        if (!threadBean.isCurrentThreadCpuTimeSupported()) {
+        if (!supported) {
             result.add(new HealthCheckResultEntry(HealthCheckResultStatus.CHECK_ERROR, "JVM implementation or OS does" +
                     " not support getting CPU times"));
             return result;
         }
+        acceptHoggingThreads(checkRecordsByThreadId, 
+                (percentage, threshold, totalTimeHogging, initialMethod, info) -> {
+                    result.add(new HealthCheckResultEntry(HealthCheckResultStatus.CRITICAL,
+                            "Thread with <id-name>: " + info.getThreadId() + "-" + info.getThreadName() +
+                            " is a hogging thread for the last " +
+                            prettyPrintDuration(totalTimeHogging) + "\n" + prettyPrintStackTrace(info.getStackTrace())));
+                });
+        return result;
+    }
 
-        final long[] ids = threadBean.getAllThreadIds();
-        for (long id : ids) {
-            if (id == Thread.currentThread().getId())
-                continue;
-            final long c = threadBean.getThreadCpuTime(id);
-            final long u = threadBean.getThreadUserTime(id);
-            ThreadInfo threadInfo = threadBean.getThreadInfo(id);
+    @Override
+    @MonitoringData(ns = "health", intervalSeconds = 4)
+    public void collect(MonitoringDataCollector collector) {
+        if (options == null || !options.isEnabled() || !supported) {
+            return;
+        }
+        AtomicInteger hoggingThreadCount = new AtomicInteger(0);
+        AtomicLong hoggingThreadMaxDuration = new AtomicLong(0L);
+        acceptHoggingThreads(colletionRecordsByThreadId, 
+                (percentage, threshold, totalTimeHogging, initialMethod, info) -> {
+                    String thread = info.getThreadName();
+                    if (thread == null || thread.isEmpty()) {
+                        thread = String.valueOf(info.getThreadId());
+                    }
+                    collector.annotate("HoggingThreadDuration", totalTimeHogging, true, //
+                            "Thread", thread, //
+                            "Usage%", String.valueOf(percentage), //
+                            "Threshold%", String.valueOf(threshold), //
+                            "Method", initialMethod, //
+                            "Exited", String.valueOf(!initialMethod.equals(getMethod(info))));
+                    hoggingThreadCount.incrementAndGet();
+                    hoggingThreadMaxDuration.updateAndGet(value -> Math.max(value, totalTimeHogging));
+                });
+        collector
+        .collect("HoggingThreadCount", hoggingThreadCount)
+        .collect("HoggingThreadDuration", hoggingThreadMaxDuration);
```

Comment:
```suggestion
                .collect("HoggingThreadDuration", hoggingThreadMaxDuration);
```
Indenting 2: The tab vs. space wars